### DISCUSSION TITLE
Allow inclusion of snippets inside a line

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#2187] [Ideas]
It would be quite nice if snippets could be used inside a line (in-between content), rather than having them on a separate line.

This would allow setups such as this (Assuming this snippet syntax would be used):
````markdown
To use this API, include the following in your configuration file:
```xml
<dependencies>
  <dependency>
    <groupId>com.example</groupId>
    <artifactId>Example</artifactId>
    <version>-8<-"api-version.txt"--</version>
  </dependency>
</dependencies>
```
````
This would allow for the version to exist within a separate file that could be updated through certain methods (I.e. some CI tasks to pull data and update the file before docs are build).

**Comment by facelessuser** [comment]
Currently, snippets expect block content to be injected. If the snippet content is indented, it intelligently indents the page content, etc.

This obviously wouldn't work for inline content. One of the reasons I've left out inline snippet content is that since we are injecting entire pages of content, it doesn't translate as well and would cause many users confusion, and, in turn, cause me many support issues I'd have to answer.

So, let's say that when inserting inline we skip indenting the inserting content to work better with inline.

Consider the following.

```
I have some

text to insert
```

```
Table | Header
----- | ------
A     | --8<--"text.txt"
B     | Inline content
```

Now I have someone complaining that this doesn't work, and it doesn't work due to the way things are inserted.

```
Table | Header
----- | ------
A     | I have some

text to insert
B     | Inline content
```

This is one of the reasons it is implemented as it currently is. You would really need to devise a whole different approach to how things are done with inline. Currently, snippets is a very simple content insertion extension that is run as a preprocessor (before most Markdown processors run).
